_posts/2023-09-19-virtual-thread-1.adoc

```diff
@@ -0,0 +1,332 @@
+
+## How can you use virtual threads in Quarkus?
+
+Using virtual threads in Quarkus is straightforward.
+You only need to use the `@RunOnVirtualThread` annotation. 
+It indicates to Quarkus to invoke the annotated method on a virtual thread instead of a regular platform thread. 
+
+This new strategy extends the _smart dispatch_ explained in the https://quarkus.io/blog/resteasy-reactive-smart-dispatch/['to block or not to block' article]. In addition to the signature, Quarkus now looks for this specific annotation.
+If your JVM does not provide virtual thread support, it does fall back to platform threads. 
+
+Let's rewrite the same example using a virtual thread:
+
+[source, java]
+----
+@Path("/greetings")
+public class VirtualThreadApp {
+
+  @RestClient RemoteService service;
+
+  @GET
+  @RunOnVirtualThread
+  public String process() {
+    // Runs on a virtual thread because the
+    // method uses the @RunOnVirtualThread annotation.
+
+    // `service` is a rest client, it executes an I/O operation
+    var response = service.greetings(); // Blocking, but this time, it 
+                                        // does neither block the carrier thread
+                                        // nor the OS thread.
+                                        // Only the virtual thread is blocked.
+	return response.toUpperCase();
+  }
+
+}
+----
+
+It's the code from the first snippet (the imperative one), but its execution model is closer to the reactive one.
+This snippet works as described because the Quarkus REST client is virtual-thread-firendly.
+
+Virtual threads in Quarkus are not limited to HTTP endpoints.
+The following snippet shows how you can process Kafka/Pulsar/AMQP messages on virtual threads:
+
+[source, java]
+----
+@Incoming("events")
+@Transactional
+@RunOnVirtualThread
+public void persistEventInDatabase(Event event) {
+  event.persist(); // Use Hibernate ORM with Panache
+}
+----
+
+## Five things you need to know before using virtual threads for everything
+
+Well, you probably see this coming.
+There is no free lunch.
+You need to know a few things before utilizing virtual threads for everything.
+These are the reasons why, currently, there is no global switch to run exclusively on virtual threads in Quarkus.
+
+### 1. Pinning 
+
+As described above, when a virtual thread executes a blocking operation, it gets unmounted from the carrier thread, preventing the carrier thread from being blocked.
+However, sometimes, the virtual thread cannot be unmounted because its state cannot be stored in the heap. 
+It happens when the thread holds a monitor lock or has a native call in the stack:
+
+[source, java]
+----
+Object monitor = new Object();
+//...
+public void aMethodThatPinTheCarrierThread() throws Exception {
+  synchronized(monitor) {
+    Thread.sleep(1000); // The virtual thread cannot be unmounted because it holds a lock, 
+                        // so the platform thread is blocked.
```

Comment:
Say "carrier thread" instead of "platform thread" here? Sounds more appropriate to me

Reply:
Fixed!